src/shared/src/types.rs

```diff
@@ -32,6 +33,43 @@ pub enum Arg {
     Upgrade,
 }
 
+#[derive(CandidType, Deserialize, Clone, Debug, Eq, PartialEq)]
+pub struct Config {
+    pub ecdsa_key_name: String,
+    // A list of allowed callers to restrict access to endpoints that do not particularly check or use the caller()
+    pub allowed_callers: Vec<Principal>,
+    pub supported_credentials: Option<Vec<SupportedCredential>>,
+    /// Root of trust for checking canister signatures.
+    pub ic_root_key_raw: Option<Vec<u8>>,
+}
+
+impl From<InitArg> for Config {
```

Comment:
can you move the impl into module `shared/src/impls`?

it might not be a standard Rust pattern, it just something I do and which we started to do in Oisy too per extension.

Reply:
Sure thing.

Reply:
Done

Reply:
Thanks